sample/sample.go

```diff
@@ -0,0 +1,181 @@
+package sample
+
+import (
+	"errors"
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/floats"
```

Comment:
Do we need this dependency? I think the code is quite straightforward https://github.com/gonum/gonum/blob/v0.15.1/floats/floats.go#L396

Reply:
We'd just have to have a copy of this: https://github.com/gonum/gonum/blob/bdcda9a453049449163d160b98285b64ec8093a1/floats/floats.go#L403-L419

It's relatively temporary as we move to the interface using tensors instead of f64s. I'm okay with either

Reply:
IIRC it's to maintain the resolution of logits. The smallest change can effect token sampling usually. @mxyng feel free to correct if wrong

Reply:
There's a dependency on `gonum` for `sampleuv` so that package will be included regardless if this uses `floats` or not